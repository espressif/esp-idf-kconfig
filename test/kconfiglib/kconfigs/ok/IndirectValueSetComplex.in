mainmenu "Test complex indirect set"

    comment "### Test set option ####"
    comment "A should be 1"

    config A
        int "A"
        default 0

    config B
        bool "B"
        default y
        set A=1 if C
        set A=2

    config C
        bool "C"
        default y

    comment "D should be E"

        config D
            string "D"
            default "D"

        config E
            bool "E"
            default y
            set D="E"

        config F # F's set will be second in line -> not used
            bool "F"
            default y
            set D="F"

    comment "G should have the same value as D"

        config G
            string "G"
            default "G"

        config H
            bool "H"
            default y
            set G=D

    comment "I should be 0x42"

        config I
            hex "I"
            default 0x00

        config J
            bool "J"
            default y
            set I=0x42 if G="E"

        config K
            bool "K"
            default y
            set I=0x43 if G="F" || G="G"
            set I=0x44


    comment "### Test set default option ###"

    comment "L should be 1"

    config L
        int "L"
        default 0

    config M
        bool "B"
        default y
        set default L=1 if N
        set default L=2

    config N
        bool "N"
        default y

    comment "O should be P"

        config O
            string "O"
            default "O"

        config P
            bool "P"
            default y
            set default O="P"

        config Q # Q's set default will be second in line -> not used
            bool "Q"
            default y
            set default O="Q"

    comment "R should have the same value as O"

        config R
            string "R"
            default "R"

        config S
            bool "S"
            default y
            set default R=O

    comment "T should be 0x42"

        config T
            hex "T"
            default 0x00

        config U
            bool "U"
            default y
            set default T=0x42 if R="P"

        config V
            bool "V"
            default y
            set default T=0x43 if R="Q" || R="R"
            set default T=0x44

    comment "### Test set and set default options simultaneously ###"

        comment "W should be X"

        config W
            string "W"
            default "W"

        config X # set default is first, but has lower precedence
            bool "X"
            default y
            set default W="Y"
            set W="X"

        comment "Y should be 0x43"
        config Y
           hex "Y"
           default 0x00

        config Z # same case as with W, but with hex value and conditions
            bool "Z"
            default y
            set default Y=0x42 if W="Y"
            set Y=0x43 if W="X" || W="W"
            set default Y=0x44
