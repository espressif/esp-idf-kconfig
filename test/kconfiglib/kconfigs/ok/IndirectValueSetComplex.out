#
# Automatically generated file. DO NOT EDIT.
# Espressif IoT Development Framework (ESP-IDF)  Project Configuration
#

#
# ### Test set option ####
#

#
# A should be 1
#
# default:
CONFIG_A=1
# default:
CONFIG_B=y
# default:
CONFIG_C=y

#
# D should be E
#
# default:
CONFIG_D="E"
# default:
CONFIG_E=y
# default:
CONFIG_F=y

#
# G should have the same value as D
#
# default:
CONFIG_G="E"
# default:
CONFIG_H=y

#
# I should be 0x42
#
# default:
CONFIG_I=0x42
# default:
CONFIG_J=y
# default:
CONFIG_K=y

#
# ### Test set default option ###
#

#
# L should be 1
#
# default:
CONFIG_L=1
# default:
CONFIG_M=y
# default:
CONFIG_N=y

#
# O should be P
#
# default:
CONFIG_O="P"
# default:
CONFIG_P=y
# default:
CONFIG_Q=y

#
# R should have the same value as O
#
# default:
CONFIG_R="P"
# default:
CONFIG_S=y

#
# T should be 0x42
#
# default:
CONFIG_T=0x42
# default:
CONFIG_U=y
# default:
CONFIG_V=y

#
# ### Test set and set default options simultaneously ###
#

#
# W should be X
#
# default:
CONFIG_W="X"
# default:
CONFIG_X=y

#
# Y should be 0x43
#
# default:
CONFIG_Y=0x43
# default:
CONFIG_Z=y
